common/src/main/scala/com/gu/recipeasy/models/Statistics.scala

```diff
@@ -52,3 +54,48 @@ object Leaderboard {
   }
 }
 
+case class UserSpeeds(curation: Double, verification: Double)
+
+object UsersSpeedsMeasurements {
+
+  def average(numbers: List[Double]): Double = {
+    if (numbers.size != 0) {
+      numbers.sum / numbers.size
+    } else {
+      0.toDouble
+    }
+  }
+
+  def secondsBetweenEvents(eventStart: UserEventDB, eventEnd: UserEventDB): Double = {
+    (eventEnd.event_datetime.getTime - eventStart.event_datetime.getTime).toDouble / 1000 // converting to seconds
+  }
+
+  def generalUserSpeedMapping(db: DB): Map[String, UserSpeeds] = {
+    val userEventsAll: List[UserEventDB] = db.userEventsAll().reverse
+    db.userEmails().map { email =>
+
+      val userEvents = userEventsAll.filter(event => event.user_email == email)
+
+      val curationEvents = userEvents.filter(event => ((event.operation_type == UserEventAccessRecipeCurationPage.name) || (event.operation_type == UserEventCuration.name)))
```

Comment:
You don't need all of the brackets so this may become:

```
val curationEvents = userEvents.filter(event => event.operation_type == UserEventAccessRecipeCurationPage.name || event.operation_type == UserEventCuration.name)
```